### ISSUE TITLE
Does tf.nn.batch_normalization support fp16

### ISSUE THREAD

**Issue by KANGRuipeng** [tensorflow/tensorflow#29071]
fp16 is new feature of Nvida GPUs, does batch_normalization support fp16?

**Comment by KANGRuipeng** [comment]
https://www.tensorflow.org/api_docs/python/tf/nn/batch_normalization, the official docs only notice " arbitrary dimensionality"

**Comment by lufol** [comment]
Running is on tensorflow cpu I got these results:
`a = np.array([5, 5, 5], dtype='float16')`
`tf.nn.batch_normalization(a, 1, 1, 0, 0, 0.001)`
Returns `<tf.Tensor 'batchnorm_5/add:0' shape=(3,) dtype=float16>`

**Comment by KANGRuipeng** [comment]
got it thks
